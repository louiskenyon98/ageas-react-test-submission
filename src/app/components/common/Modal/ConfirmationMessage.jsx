import React from 'react';
import styles from '../../../../themes/style.scss';

export default class ConfirmationMessage extends React.PureComponent {
    render() {
        return (
            <React.Fragment>
                <h1>Is you sure fam?</h1>
                {this.props.body}
                <button className={`${styles.modalButton} ui inverted primary button`}
                        onClick={this.props.accept.callback}>{this.props.accept.label}
                </button>
            </React.Fragment>
        )
    }
}
