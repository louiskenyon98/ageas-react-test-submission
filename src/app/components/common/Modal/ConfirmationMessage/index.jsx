import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../../../../themes/style.scss';

export default class ConfirmationMessage extends React.PureComponent {
    render() {
        return (
            <React.Fragment>
                <h1>{this.props.title}</h1>
                {this.props.body}
                <div style={{float: 'right' , margin: 20}}>
                    {this.props.decline && <button
                        className={`${styles.modalButton} ui inverted primary button`}
                        // onClick={this.props.accept.callback}>{this.props.accept.label}
                        onClick={this.props.decline.onClick}
                    >
                        {this.props.decline.label}
                    </button>}
                    {this.props.accept && <button
                        className={`${styles.modalButton} ui inverted primary button`}
                        // onClick={this.props.accept.callback}>{this.props.accept.label}
                        onClick={this.props.accept.onClick}
                    >
                        {this.props.accept.label}
                    </button>}
                </div>
            </React.Fragment>
        )
    }
}

ConfirmationMessage.propTypes = {
    title: PropTypes.string.isRequired,
    body: PropTypes.string,
    decline: PropTypes.shape({
        label: PropTypes.string.isRequired,
        onClick: PropTypes.func.isRequired
    }),
    accept: PropTypes.shape({
        label: PropTypes.string.isRequired,
        onClick: PropTypes.func.isRequired
    }),
};
