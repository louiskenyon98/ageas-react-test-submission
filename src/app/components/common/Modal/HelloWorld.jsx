import React from 'react';

export default class HelloWorld extends React.PureComponent {
    render() {
        return (
            <div>
                {this.props.message}
            </div>
        )
    }
}
