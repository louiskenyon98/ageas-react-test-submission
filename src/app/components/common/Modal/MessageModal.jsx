import React from 'react';
//:todo rename messagemodal to message, and also create errormessage component
export default class MessageModal extends React.PureComponent {
    render() {
        return (
            <div>
                {this.props.props.message}
            </div>
        )
    }
}
