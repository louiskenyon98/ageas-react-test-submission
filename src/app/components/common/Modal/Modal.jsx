import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../../../themes/style.scss';

class Modal extends React.Component {
    render() {
        if (this.props.show) {
            return (
                <React.Fragment>
                    <div className={styles.veil} onClick={this.props.close}/>
                    <div className={styles.modal}>
                        {this.props.title && <h1>{this.props.title}</h1>}
                        {this.props.children}
                    </div>
                </React.Fragment>
            )
        }
        return null;
    }

}

Modal.propTypes = {
    show: PropTypes.bool,
    close: PropTypes.func,
    title: PropTypes.string,
    children: PropTypes.object
};

export default Modal;
