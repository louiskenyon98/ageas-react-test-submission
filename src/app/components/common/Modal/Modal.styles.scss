.veil {
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  //background: rgba(84, 200, 255, 0.64);
  background: rgba(0, 0, 0, 0.37);
  z-index: 1;
  position: fixed;
}
.modal {
  h1 {
    font-family: 'Thasadith', sans-serif !important;
  }
  z-index: 2;
  height: auto;
  background-color: white;
  box-shadow: 6px 6px 8px 1px;
  padding: 2vh;
  position: fixed;
  width:60%;
  top: 50%;
  left: 50%;
  border-radius: .28571429rem;
  transform: translate(-50%, -50%);
  .modalButton {
    z-index: 3;
    float: right;
  }
}
