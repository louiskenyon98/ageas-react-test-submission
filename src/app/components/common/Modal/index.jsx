import React from 'react';
import styles from '../../../../themes/style.scss';

class Modal extends React.PureComponent {
    render() {
        const {show, children, closeModal} = this.props;
        //1 is placeholder for visible prop.
        if (show) {
            return (
                <React.Fragment>
                    <div className={styles.veil}/>
                    <div className={styles.modal}>
                        {children}
                        {/*<button className={styles.modalButton} onClick={closeModal}>close</button>*/}
                        <button className={`${styles.modalButton} ui inverted primary button`} onClick={closeModal}>close</button>
                    </div>
                </React.Fragment>
            )
        }
        return null;
    }

}

export default Modal;
