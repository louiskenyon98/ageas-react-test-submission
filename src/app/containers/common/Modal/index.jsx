import React from 'react';
import {connect} from 'react-redux';
import {closeModal} from '../../../actions/common/modal.action';
import Modal from '../../../components/common/Modal/Modal';
import StatusMessage from "../../../components/common/Modal/StatusMessage/Index";
import DeleteNoteContainer from "./ConfirmationMessage/DeleteNoteContainer";

export class ModalContainer extends React.Component {
    returnModal() {
        switch (this.props.modal.type) {
            case 'status':
                return (
                    <StatusMessage
                        {...this.props.modal.props}
                        close={this.props.closeModal}
                    />
                );
            case 'deleteNoteConfirmation':
                return (
                    <DeleteNoteContainer
                        {...this.props.modal.props}
                    />
                );
        }
    }

    render() {
        return (
            <Modal
                show={this.props.modal.show}
                close={this.props.closeModal}
            >
                {this.returnModal()}
            </Modal>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        modal: state.common.modal
    }
};

export default connect(mapStateToProps, {closeModal})(ModalContainer)
