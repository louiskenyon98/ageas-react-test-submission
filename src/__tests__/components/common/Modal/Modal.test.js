import React from 'react';
import {shallow} from 'enzyme';
import Modal from '../../../../app/components/common/Modal/Modal';

describe('Modal', () => {
    let props = {};
    let wrapper = () => shallow(<Modal {...props}/>);

    beforeEach(() => {
        props = {
            close: jest.fn(),
            show: true,
            children: <div>Hello world</div>,
            title: 'This is a modal title',
        }
    });

    describe('render', () => {
        it('should render a component with children', () => {
            expect(wrapper()).toMatchSnapshot();
        });

        it('should not render when show equals false', () => {
            props.show = false;
            expect(wrapper()).toMatchSnapshot();
        });

        it('should not render a title when one is not passed in', () => {
            props.title = '';
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        it('should call close onClick', () => {
            expect(props.close).not.toHaveBeenCalled();
            expect(props.close).toHaveBeenCalledTimes(0);
            wrapper().find('button').first().simulate('click');
            expect(props.close).toHaveBeenCalledTimes(1);
        })
    });
});
