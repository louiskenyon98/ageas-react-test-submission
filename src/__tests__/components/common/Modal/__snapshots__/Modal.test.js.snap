// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should not render a title when one is not passed in 1`] = `
<Fragment>
  <div
    className="veil"
  />
  <div
    className="modal"
  >
    <div>
      Hello world
    </div>
    <button
      className="modalButton ui inverted red button"
      onClick={[MockFunction]}
    >
      close
    </button>
  </div>
</Fragment>
`;

exports[`Modal should not render when show equals false 1`] = `""`;

exports[`Modal should render a component with children 1`] = `
<Fragment>
  <div
    className="veil"
  />
  <div
    className="modal"
  >
    <h1>
      This is a modal title
    </h1>
    <div>
      Hello world
    </div>
    <button
      className="modalButton ui inverted red button"
      onClick={[MockFunction]}
    >
      close
    </button>
  </div>
</Fragment>
`;
